replica/rebase.go

```diff
@@ -26,12 +26,13 @@ type BlockIterator interface {
 }
 
 type Validator interface {
-	IsBlockValid(block block.Block, checkHistory bool, shard Shard) error
+	IsBlockValid(block block.Block, checkHistory bool, shard Shard) (process.NilReasons, error)
 }
 
 type Observer interface {
 	DidCommitBlock(block.Height, Shard)
 	IsSignatory(Shard) bool
+	DidReceiveSufficientNilPrevotes(messages process.Messages, threshold int)
```

Comment:
For aesthetic purposes, can we please put the two `Did` functions next to each other.